### PR TITLE
[feat] added some voice functions

### PR DESCRIPTION
**Pull request by alnitak**
## Description

- added:
  - `getActiveVoiceCount()` to get concurrent sounds that are playing at the moment.
  - `getCountAudioSource()` to get concurrent sounds that are playing a specific audio source.
  - `getVoiceCount()` to get the number of voices the application has told SoLoud to play.
  - `getProtectVoice()`
  - `setProtectVoice()`
  - `getMaxActiveVoiceCount()` to get the current maximum active voice count.
  - `setMaxActiveVoiceCount()` to set the current maximum active voice count.

Seem that `setProtectVoice()` doesn't work as expected. I tried [this fix](https://github.com/jarikomppa/soloud/issues/298) but doesn't work for me. 
Is `setMaxActiveVoiceCount()` enough for #53 ?

For now I comment out `getProtectVoice()` and `setProtectVoice()`, I'll do some more research.

## Type of Change

- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: lib/src/soloud.dart [hunk 1]
```diff
@@ -1432,6 +1432,84 @@ interface class SoLoud {
...
+  // /// If all voices are protected, the result will be undefined.
+  // ///
+  // /// [handle]  handle to check.
+  // /// [protect] whether to protect or not.
+  // void setProtectVoice(int handle, bool protect) {
+  //   if (!isInitialized) {
+  //     throw const SoLoudNotInitializedException();
+  //   }
+  //   SoLoudController().soLoudFFI.setProtectVoice(handle, protect);
+  // }
```
Comment (filiph) [inline lib/src/soloud.dart:1483]: Why are these commented out?

Comment (alnitak) [reply lib/src/soloud.dart:1483]: I wrote that in the description. It seems that `setProtectVoice` does nothing as stated in [this issue](https://github.com/jarikomppa/soloud/issues/298). I tried that fix, but it doesn't work for me. I have to do some more research and tests.

Anyway `setMaxActiveVoiceCount` does its job. The default max voices are set to 16 by default but can be raised to 4095. Maybe 16 is a low max number for games, we may think about raising it when initializing the player or let the devs set it while calling `initialize`.

`setProtectVoice()` just tells the player to continue to play that sound even if the max active voice count is reached (ie a background music). The default value of 16 is for performance. The more concurrent voices are playing, the more CPU power is needed. So it's also up to devs to check this.

This is explained better [here](https://solhsa.com/soloud/soloud_20200207.html#voice).

File: src/player.cpp [hunk 2]
```diff
@@ -438,11 +438,59 @@ void Player::setVolume(SoLoud::handle handle, float volume)
...
+{
+    auto const &s = std::find_if(sounds.begin(), sounds.end(),
+                                 [&](std::unique_ptr<ActiveSound> const &f)
+                                 { return f->soundHash == soundHash; });
+    if (s == sounds.end() || s->get()->soundType == TYPE_SYNTH)
+        return 0;
+    if (s->get()->soundType == TYPE_WAV)
+    {
+        SoLoud::AudioSource *as = static_cast<SoLoud::Wav *>(s->get()->sound.get());
+        return soloud.countAudioSource(*as);
```
[the remark this answers is not in the document]
Comment (alnitak) [reply src/player.cpp]: Renamed to `countAudioSource`

File: example/lib/page_visualizer.dart [hunk 3]
```diff
@@ -52,37 +53,32 @@ class _PageVisualizerState extends State<PageVisualizer> {
...
+                  SoLoudCapture.instance.startCapture();
+                  visualizerController.changeIsCaptureStarted(true);
+                }
+              },
+              onDeviceIdChanged: (deviceID) {
+                SoLoudCapture.instance.stopCapture();
+                SoLoudCapture.instance.initialize(deviceID: deviceID);
+                SoLoudCapture.instance.startCapture();
+              },
+            ),
```
[the remark this answers is not in the document]
Comment (alnitak) [reply example/lib/page_visualizer.dart:79]: Ok, we will think about some capture examples later.

**Pull request merged**